tests/tracer/test_sampler.py

```diff
@@ -261,6 +261,62 @@ def test_sampling_rule_init():
     assert rule.name == name_regex
 
 
+def test_sampling_rule_init_via_env():
+    # Testing single sampling rule
+    with override_env(dict(DD_TRACE_SAMPLING_RULES='[{"sample_rate":1.0,"service":"xyz","name":"abc"}]')):
+        sampling_rule = DatadogSampler().rules
+        assert sampling_rule[0].sample_rate == 1.0
+        assert sampling_rule[0].service == "xyz"
+        assert sampling_rule[0].name == "abc"
+
+    # Testing multiple sampling rules
+    with override_env(
+        dict(
+            DD_TRACE_SAMPLING_RULES='[{"sample_rate":1.0,"service":"xyz","name":"abc"}, \
+            {"sample_rate":0.5,"service":"my-service","name":"my-name"}]'
+        )
+    ):
+        sampling_rule = DatadogSampler().rules
+        assert sampling_rule[0].sample_rate == 1.0
+        assert sampling_rule[0].service == "xyz"
+        assert sampling_rule[0].name == "abc"
+
+        assert sampling_rule[1].sample_rate == 0.5
+        assert sampling_rule[1].service == "my-service"
+        assert sampling_rule[1].name == "my-name"
+
+    # Testing for only Sample rate being set
+    with override_env(dict(DD_TRACE_SAMPLING_RULES='[{"sample_rate":1.0}]')):
+        sampling_rule = DatadogSampler().rules
+        assert sampling_rule[0].sample_rate == 1.0
+        assert sampling_rule[0].service == SamplingRule.NO_RULE
+        assert sampling_rule[0].name == SamplingRule.NO_RULE
+
+    # Testing for no name being set
+    with override_env(dict(DD_TRACE_SAMPLING_RULES='[{"sample_rate":1.0,"service":"xyz"}]')):
+        sampling_rule = DatadogSampler().rules
+        assert sampling_rule[0].sample_rate == 1.0
+        assert sampling_rule[0].service == "xyz"
+        assert sampling_rule[0].name == SamplingRule.NO_RULE
+
+    # Testing for no service being set
+    with override_env(dict(DD_TRACE_SAMPLING_RULES='[{"sample_rate":1.0,"name":"abc"}]')):
+        sampling_rule = DatadogSampler().rules
+        assert sampling_rule[0].sample_rate == 1.0
+        assert sampling_rule[0].service == SamplingRule.NO_RULE
+        assert sampling_rule[0].name == "abc"
+
+    # Testing for only Sample rate greater than 1.0
+    with override_env(dict(DD_TRACE_SAMPLING_RULES='[{"sample_rate":2.0,"service":"xyz","name":"abc"}]')):
+        sampling_rule = DatadogSampler().rules
+        assert sampling_rule == []
+
+    # Testing for no Sample rate
+    with override_env(dict(DD_TRACE_SAMPLING_RULES='[{"service":"xyz","name":"abc"}]')):
+        sampling_rule = DatadogSampler().rules
+        assert sampling_rule == []
```

Comment:
based on past comments, we should validate that these raise exceptions.

e.g.

```python
with pytest.raises(ValueError):  # or whichever exception we are raising
    with override_env():
        DatadogSampler()
```

Reply:
👍